src/components/common/FileSelect/FileSelect.js

```diff
@@ -148,44 +132,43 @@ class FileSelect extends Component {
     }
   }
 
-  getRawData (file) {
-    return new Promise(resolve => {
-      const reader = new window.FileReader()
-      reader.onload = () => resolve(reader.result)
-      // TODO: use array buffers instead of base64 strings
-      reader.readAsDataURL(file)
-    })
+  handleFileUpdate = (file: FileModel) => {
+    this.setState((prevState) => ({
+      fileCollection: prevState.fileCollection.update(file)
+    }))
   }
 
-  handleChange = async (e) => {
-    this.setState({
-      isLoading: true,
-      isLoaded: false,
-      error: null
-    })
-    const uploadedFiles = e.target.files || []
-    const files = []
+  async uploadCollection (files: FileCollection, config: fileConfig) {
+    const fileCollection  = await ipfs.uploadCollection(files, config, this.handleFileUpdate)
+    this.setState({fileCollection})
+    this.props.input.onChange(fileCollection.hash())
+  }
+
+  getFilesLeft () {
+    return Math.max(this.state.config.maxFiles - this.state.fileCollection.size(), 0)
+  }
 
+  handleChange = async (e) => {
+    if (!e.target.files.length) {
+      return
+    }
+    const {config} = this.state
+    const {multiple} = this.props
+    let fileCollection = multiple
+      ? this.state.fileCollection
+      : new FileCollection()
+    fileCollection = fileCollection.uploading(true)
+    let fileModel
+    const uploadedFiles = [...e.target.files].slice(0, this.getFilesLeft())
     for (let file of uploadedFiles) {
-      try {
-        let parsedFile = await this.parseFile(file)
-        let rawData = await this.getRawData(file)
-        let isLoaded = await this.addToIPFS(file.name, rawData)
-        if (isLoaded) {
-          files.push(parsedFile)
-        }
-      } catch (e) {
-        this.setState({
-          error: e.message,
-          isLoading: false
-        })
-        // TODO @bshevchenko: Improve FileSelect
-        // eslint-disable-next-line
-        console.error(e)
-      }
+      fileModel = new FileModel({
+        file,
+        uploading: true
+      })
+      fileCollection = fileCollection.add(fileModel)
     }
-
-    this.setState({files})
+    this.setState({fileCollection})
+    await this.uploadCollection(fileCollection, config)
   }
 
   handleOpenFileDialog = () => {
```

Comment:
remove unused this.state.isLoading

Reply:
It is useful, it disables 'Add Attachment' button instead of colleciton.uploading(), because the uploading flag in the FileCollections works more like dirty flag, sometimes collection is dirty because usr tried to upload invalid file.